client/app/scripts/liveblog-advertising/controllers/advertising.js

```diff
@@ -0,0 +1,241 @@
+        $scope.freetypeControl.reset();
+        $scope.advertModalActive = false;
+        $scope.dialogAdvertLoading = false;
+        return loadAdverts(true);
+    }
+
+    function handleAdvertSaveError() {
+        notify.error(gettext('Something went wrong, please try again later!'), 5000)
+    }
+
+    $scope.removeAdvert = function (advert, $index) {
+        modal.confirm(gettext('Are you sure you want to remove this advert?')).then(function() {
+            api('advertisements').save(advert, {deleted: true})
+            .then(function(data) {
+                $scope.adverts.splice($index, 1);
+            }, function(data) {
+                notify.error(gettext('Can\'t remove advert'));
+            });
+        });
+    }
+
+    $scope.saveAdvert = function() {
+        var newAd = {
+            name: $scope.advert.name,
+            type: $scope.advertType.name,
+            text: freetypeService.htmlContent($scope.advertType.template, $scope.freetypesData),
+            meta: {data: $scope.freetypesData}
+        }
+        $scope.dialogAdvertLoading = true;
+
+        api('advertisements').save($scope.advert, newAd)
+        .then(handleAdvertSaveSuccess, handleAdvertSaveError);
+    }
+
+    loadAdverts();
+
+
+    //COLLECTIONS
+    function loadCollections(silent) {
+        silent = silent || false;
+        if (!silent) {
+            $scope.collectionsLoading = true;
+        }
+        return api('collections').query({where: {deleted: false}})
+        .then(function(data) {
+            $scope.collections = data._items;
+            if (!silent) {
+                notify.info(gettext('Collections loaded'));
+            }
+            $scope.collectionsLoading = false;
+        })
+        .catch(function(data) {
+            $scope.collectionsLoading = false;
+            notify.error(gettext('There was an error getting the adverts'));
+        })
+    }
+
+    function handleCollectionSaveSuccess() {
+        notify.info(gettext('Collection saved successfully'));
+        $scope.collection = {};
+        $scope.collectionModalActive = false;
+        $scope.dialogCollectionLoading = false;
+        return loadCollections(true);
+    }
+
+    function handleCollectionSaveError() {
+        notify.error(gettext('Something went wrong, please try again later!'), 5000)
+    }
+
+    $scope.openCollectionDialog = function(collection) {
+        collection = collection || false;
+        // load all available adverts without showing any messages
+        loadAdverts(true).then(function() {
+            if (collection) {
+                // editing collection
+                $scope.collection = angular.copy(collection);
+                $scope.collection.checkAdverts = {};
+                //console.log('$scope.collection ', $scope.collection);
+                angular.forEach($scope.adverts, function(advert) {
+                    if ($scope.collectionHasAdvert($scope.collection, advert)) {
```

Comment:
Can you not just write this block in one line? You don't need an if/else to set a boolean